packages/components/src/app/BaseApp.tsx

```diff
@@ -126,14 +139,15 @@ function BaseApp({
     }
   }, [handleConsentGiven, asPath]);
 
-  const askemFeedbackInstance = React.useMemo(
-    () =>
-      createAskemInstance({
-        ...askemFeedbackConfiguration,
-        consentGiven: askemConsentGiven,
-      }),
-    [askemFeedbackConfiguration, askemConsentGiven]
-  );
+  const newAskemConfiguration: AskemConfigs = {
+    ...askemConfigurationInput,
+    consentGiven: askemConsentGiven,
+  };
+
+  if (!askemInstance || !isEqual(askemConfiguration, newAskemConfiguration)) {
+    setAskemConfiguration(newAskemConfiguration);
+    setAskemInstance(createAskemInstance(newAskemConfiguration));
+  }
```

Comment:
I don't get this. Handling this wit a state should not make a difference compared to a memo. The memo is just a lot cleaner. Both are executed when the dependency array changes.

I need to test this. How was the issue reproducable?

Reply:
Found the instructions from the PR description 😝 

```
Testing
checkout PR's branch locally because review environments don't have Askem enabled
cd apps/sports-helsinki
yarn
yarn dev
open page https://localhost:3000/fi/articles/yleinen/testiartikkeli-syyskuu in browser
make sure Askem cookies are allowed, if you're not sure clear localhost:3000's cookies or at least the city-of-helsinki-cookie-consents cookie
do twice in a row:
scroll down to the bottom of the page
scroll back up to the top of the page
reload
What is expected:

Everything should be fine and the network shouldn't be loading anything in a seemingly infinite loop.
Askem functionality should also work all around correctly.
```

Reply:
useMemo with any object variables is relatively useless IMO as it doesn't compare the objects' contents but simply the references themselves → useMemo always changes the return value if any of useMemo's dependency list's objects' references change even though their contents stay the same. I added deep equality checking to get around this.

Reply:
It actually can compare deeply: the dependency array could be something like `[JSON.stringify(object)]`. As a reference I found this for example: https://javascript.plainenglish.io/a-less-known-trick-with-react-hook-dependencies-8dafaca7a150

Reply:
According to [useMemo documentation](https://react.dev/reference/react/useMemo#memoizing-a-dependency-of-another-hook) it seems they're suggesting to memoize objects used as useMemo dependencies → recursively going back to only using primitive types as dependencies in useMemo in the leaf nodes of the useMemo calls.

The [JSON.stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) has some limitations:
- BigInt -> throw
- undefined -> null
- Function -> nothing or null
- Symbol -> nothing or null
- NaN -> null
- Infinity -> null
- Map -> "{}"
- Set -> "{}"

which could possibly be overcome with the use of the [replacer ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter)parameter.